lit_llama/adapter.py

```diff
@@ -2,6 +2,41 @@
 
 LLaMA-Adapter: Efficient Fine-tuning of Language Models with Zero-init Attention
 https://arxiv.org/abs/2303.16199
+
+                                                                             |              Prefix cross-attention
+                                                                             |
+  ┌─────────────────┐                                                        |               ┌──────────────────┐
+  ┆        x        ┆                                                        |               ┆      prefix      ┆
+  └─────────────────┘                                                        |               └──────────────────┘
+           |                                                                 |                        |
+           ▼                                                                 |                        ▼
+  ┌──────────────────┐                                                       |              ┌─────────────────────┐
+  ┆  self-attention  ┆ --------------------------------------------------------------┐      ┆  linear projection  ┆
```

Comment:
This looks great! And the write-up is also great! Thanks a lot for doing this!

Reply:
So pleased to hear it. Thanks a lot! 😄 

![](https://media.giphy.com/media/hMYam42K5GSGjSnUpz/giphy-downsized-large.gif)

Reply:
lol